### PR TITLE
Additions from GMAO (2): Add hooks to get melt_potential and OBLD

### PR DESCRIPTION
**Pull request by sanAkel**
This PR includes addition of:
- melt potential
- ocean boundary layer depth (OBLD)

to variables that can obtained by `ocean_model_data2D_get`

### CODE REVIEW

File: config_src/coupled_driver/ocean_model_MOM.F90 [hunk 1]
```diff
@@ -802,6 +806,8 @@ subroutine initialize_ocean_public_type(input_domain, Ocean_sfc, diag, maskmap,
   Ocean_sfc%v_surf(:,:)  = 0.0  ! time averaged v-current (m/sec)  passed to atmosphere/ice models
   Ocean_sfc%sea_lev(:,:) = 0.0  ! time averaged thickness of top model grid cell (m) plus patm/rho0/grav
   Ocean_sfc%frazil(:,:)  = 0.0  ! time accumulated frazil (J/m^2) passed to ice model
+  Ocean_sfc%melt_potential  = 0.0  ! time accumulated melt potential (J/m^2) passed to ice model
```
Comment (Hallberg-NOAA) [inline config_src/coupled_driver/ocean_model_MOM.F90:809]: This is an array syntax initialization statement. The MOM6 code style guide ( https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide ) requires the addition of '(:,:)' to make it clear that this is this initialization of a 2-d array, so this line should become `Ocean_sfc%melt_potential(:,:) = 0.0 ! ...' , and similarly on the next line.

**Review by Hallberg-NOAA — APPROVED** [review]
These changes seem reasonable and properly implemented.  I think that they are in line be merged into MOM6 after the TC and regression testing have passed.

**Comment by sanAkel** [comment]
> These changes seem reasonable and properly implemented. I think that they are in line be merged into MOM6 after the TC and regression testing have passed.

Thank you @Hallberg-NOAA , I was thinking of adding an _optional_ arg to ocean_model_init as you suggested for the stress https://github.com/NOAA-GFDL/MOM6/blob/80a6df4a0e232756ea02b3c71c9b92786798c242/config_src/coupled_driver/ocean_model_MOM.F90#L237 via `HFREEZE` from our _coupler_  https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/develop/GEOSogcm_GridComp/GEOSocean_GridComp/GuestOcean_GridComp/MOM6_GEOSPlug/MOM6_GEOSPlug.F90 

But given the current status of our ice-ocean coupling (in GEOS) backed away! However, I am thinking how you handle these fields (melt_potential and OBLD) with SIS2. Will send an email to continue this conversation _offline_.

**Comment by marshallward** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/11572

**Pull request merged**